core/domain/classifier_domain_test.py

```diff
@@ -269,3 +269,59 @@ def test_validation(self):
         with self.assertRaisesRegexp(utils.ValidationError, (
             'Expected training_data to be a list')):
             training_job.validate()
+
+
+class ClassifierExplorationMappingDomainTests(test_utils.GenericTestBase):
+    """Tests for the ClassifierExplorationMapping domain."""
+
+    def _get_mapping_from_dict(self, mapping_dict):
+        mapping = classifier_domain.ClassifierExplorationMapping(
+            mapping_dict['exp_id'],
+            mapping_dict['exp_version'],
+            mapping_dict['state_name'],
+            mapping_dict['classifier_id'])
+
+        return mapping
+
+    def test_to_dict(self):
+        expected_mapping_dict = {
+            'exp_id': 'exp_id1',
+            'exp_version': 2,
+            'state_name': 'state_name1',
+            'classifier_id': 'classifier_id1'
+        }
+        observed_mapping = self._get_mapping_from_dict(
+            expected_mapping_dict)
+        self.assertDictEqual(expected_mapping_dict,
+                             observed_mapping.to_dict())
+
+    def test_validation(self):
```

Comment:
@anmolshkl As we discussed with Sean, the state_name does not need to be unicode (depends on what the creator specifies, it might be just ascii). If it turns out to be unicode, we convert it to a string by encoding to utf-8. So, more or less, we never work with unicode. If we encounter it, we convert it. Is that clear?